core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -271,6 +271,9 @@ public boolean requiresAnalysisSettings() {
             Settings currentSettings = entry.getValue();
             String typeName = currentSettings.get("type");
             if (analyzer) {
+                if(defaultInstance.containsKey(name)) {
```

Comment:
I think we should do this for all analysis components, so I would move this test outside of the `if (analyzer)` block. Can you also add one space between your if and the opening parenthesis so that indentation is consistent with the rest of the code base?